presto-test-jdbc-compatibility-old-server/src/test/java/io/prestosql/TestJdbcResultSetCompatibility.java

```diff
@@ -0,0 +1,125 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import io.airlift.log.Logging;
+import io.prestosql.jdbc.BaseTestJdbcResultSet;
+import io.prestosql.jdbc.PrestoDriver;
+import org.testcontainers.containers.PrestoContainer;
+import org.testng.SkipException;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+@Test(singleThreaded = true)
+public class TestJdbcResultSetCompatibility
+        extends BaseTestJdbcResultSet
+{
+    private static final int NUMBER_OF_TESTED_VERSIONS = 5;
+    private static final int TESTED_VERSIONS_GRANULARITY = 3;
+
+    private final Optional<String> testedPrestoVersion;
+    private PrestoContainer<?> prestoContainer;
+
+    @Factory(dataProvider = "testedPrestoVersions")
+    public TestJdbcResultSetCompatibility(Optional<String> testedPrestoVersion)
+    {
+        this.testedPrestoVersion = requireNonNull(testedPrestoVersion, "testedPrestoVersion is null");
+    }
+
+    @DataProvider
+    public static Object[][] testedPrestoVersions()
+    {
+        try {
+            String currentVersionString = PrestoDriver.class.getPackage().getImplementationVersion();
+            if (currentVersionString == null) {
+                return new Object[][] {
+                        {Optional.empty()}
+                };
+            }
+
+            int currentVersion = Integer.parseInt(Splitter.on("-").omitEmptyStrings().trimResults().splitToList(currentVersionString).get(0));
+            ImmutableList.Builder<String> testedPrestoVersions = ImmutableList.builder();
+            int testedVersion = currentVersion - 1;
+            for (int i = 0; i < NUMBER_OF_TESTED_VERSIONS; i++) {
+                testedPrestoVersions.add(String.valueOf(testedVersion));
+                testedVersion -= TESTED_VERSIONS_GRANULARITY;
+            }
+
+            return testedPrestoVersions.build().stream()
+                    .map(version -> new Object[] {Optional.of(version)})
+                    .toArray(Object[][]::new);
+        }
+        catch (Throwable e) {
+            System.err.println("Could not determine Presto versions to test; " + e.getMessage() + "\n" + getStackTrace(e));
```

Comment:
Let it fly!

Reply:
I can't. It kills VM. We do not want that.

Reply:
Did the stacktrace get obliterated?

Reply:
I am pretty sure it is. Otherwise I would not be that concerned.

Reply:
I checked. It is obiterated. The exception message is all that remains.

Reply:
Add a comment. Otherwise `System.err.println(` looks like a debug / leftover.

Reply:
There is a comment here already at newest version :) I think you requested it previosly.

Reply:
Damn - just noticed I did not push it....